src/org/zaproxy/zap/extension/tlsdebug/TlsDebugPanel.java

```diff
@@ -0,0 +1,252 @@
+			@Override
+			public void actionPerformed(java.awt.event.ActionEvent e) {
+				NodeSelectDialog nsd = new NodeSelectDialog(View.getSingleton().getMainFrame());
+				SiteNode node = null;
+				try {
+					node = Model.getSingleton().getSession().getSiteTree()
+							.findNode(new URI(getUrlField().getText(), false));
+				} catch (Exception e2) {
+					// Ignore
+				}
+				node = nsd.showDialog(node);
+				if (node != null && node.getHistoryReference() != null) {
+					try {
+						getUrlField().setText(node.getHistoryReference().getURI().toString());
+					} catch (Exception e1) {
+						// Ignore
+					}
+				}
+			}
+		});
+
+		urlSelectPanel.add(this.getUrlField(), LayoutHelper.getGBC(0, 0, 1, 1.0D));
+		urlSelectPanel.add(selectButton, LayoutHelper.getGBC(1, 0, 1, 0.0D));
+		panelContent.add(urlSelectPanel, LayoutHelper.getGBC(1, 0, 3, 0.25D));
+
+		panelContent.add(this.getCheckButton(), LayoutHelper.getGBC(0, 1, 1, 0.0D));
+
+		JPanel outputPanel = new JPanel(new BorderLayout());
+		outputPanel.add(new JLabel(Constant.messages.getString("tlsdebug.label.console")), BorderLayout.NORTH);
+		JScrollPane jScrollPane = new JScrollPane();
+		jScrollPane.add(getOutputArea(), LayoutHelper.getGBC(0, 0, 4, 1.D, 1.0D)); // Padding
+																					// at
+																					// bottom
+		jScrollPane.setFont(new java.awt.Font("Dialog", java.awt.Font.PLAIN, 11));
+		jScrollPane.setHorizontalScrollBarPolicy(javax.swing.JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+		jScrollPane.setVerticalScrollBarPolicy(javax.swing.JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED);
+		jScrollPane.setViewportView(getOutputArea());
+		outputPanel.add(jScrollPane, BorderLayout.CENTER);
+
+		this.add(outputPanel, BorderLayout.CENTER);
+	}
+
+	private ZapTextField getUrlField() {
+		if (urlField == null) {
+			urlField = new ZapTextField();
+		}
+		return urlField;
+	}
+
+	private JButton getCheckButton() {
+		if (checkButton == null) {
+			checkButton = new JButton();
+			checkButton.setText(Constant.messages.getString("tlsdebug.button.label.check"));
+			checkButton.setIcon(DisplayUtils.getScaledIcon(TLSDEBUG_ICON));
+			// icon
+			checkButton.setToolTipText(Constant.messages.getString("tlsdebug.button.tooltip.check"));
+
+			checkButton.addActionListener(new java.awt.event.ActionListener() {
+				@Override
+				public void actionPerformed(java.awt.event.ActionEvent e) {
+					try {
+						extension.launchDebug(new URL(getUrlField().getText()));
+					} catch (MalformedURLException mue) {
+						logger.warn(mue.getMessage(), mue);
+						View.getSingleton().showWarningDialog(mue.getMessage());
+					} catch (IOException ioe) {
+						logger.warn(ioe.getMessage(), ioe);
+						View.getSingleton().showWarningDialog(ioe.getMessage());
+					}
+				}
+			});
+		}
+		return checkButton;
+	}
+
+	protected JTextArea getOutputArea() {
+		if (outputArea == null) {
+			outputArea = new JTextArea();
+
+			MouseListener popupListener = new PopupListener(getOutputAreaPopupMenu());

```

Comment:
I'd suggest using `outputArea.setComponentPopupMenu(getOutputAreaPopupMenu());` instead, that allows to access the context menu with the keyboard and mouse.